samples/received-text-processing/src/main.cpp

```diff
@@ -0,0 +1,65 @@
+#include <csignal>
+#include <cstdio>
+#include <cstdlib>
+#include <exception>
+#include <string>
+
+#include <tgbot/tgbot.h>
+
+using namespace std;
+using namespace TgBot;
+
+vector<string> bot_commands = {"start", "test"};
+
+int main() {
+    string token(getenv("TOKEN"));
+    printf("Token: %s\n", token.c_str());
+
+    bool get_test_text = false;
+    string test_text;
+
+    Bot bot(token);
+    TgLongPoll long_poll(bot);
+
+    bot.getEvents().onCommand("start", [&bot](Message::Ptr message) {
+        bot.getApi().sendMessage(message->chat->id, "Hi!");
+    });
+
+    bot.getEvents().onCommand("test", [&](Message::Ptr message) {
+        bot.getApi().sendMessage(message->chat->id, "Enter text");
+
+        get_test_text = true;
+        long_poll.start();
+        bot.getApi().sendMessage(message->chat->id, test_text);
+        get_test_text = false;
+    });
+
+    bot.getEvents().onAnyMessage([&](TgBot::Message::Ptr message) {
+        if (get_test_text) {
+            test_text = message->text;
+            return;
+        }
+
+        for (const auto& command : bot_commands) {
+            if ("/" + command == message->text) {
+                return;
+            }
+        }
+
+        bot.getApi().sendMessage(message->chat->id, "unknown command");
+    });
```

Comment:
Let long_poll.start() be invoked only in the main loop.

```suggestion
    bool test_text_state = false;

    Bot bot(token);
    TgLongPoll long_poll(bot);

    bot.getEvents().onCommand("start", [&bot](Message::Ptr message) {
        bot.getApi().sendMessage(message->chat->id, "Hi!");
    });

    bot.getEvents().onCommand("test", [&](Message::Ptr message) {
        bot.getApi().sendMessage(message->chat->id, "Enter text");
        test_text_state = true;
    });

    bot.getEvents().onAnyMessage([&](TgBot::Message::Ptr message) {
        if (test_text_state) {
            bot.getApi().sendMessage(message->chat->id, message->text);
            test_text_state = false;
            return;
        }

        for (const auto& command : bot_commands) {
            if ("/" + command == message->text) {
                return;
            }
        }

        bot.getApi().sendMessage(message->chat->id, "unknown command");
    });
```

Reply:
? you have now changed the code it will not work as it should.

Reply:
the point of all this is the second call to long_poll to get the text from the X input.